server/handlers/changefeed.go

```diff
@@ -67,15 +67,19 @@ func checkChangefeedInputs(logger ctxu.Logger, s interface{}, r string) (
 		err = errors.ErrNoStorage.WithDetail(nil)
 		return
 	}
-	pageSize, err = strconv.ParseInt(r, 10, 32)
-	if err != nil {
-		logger.Errorf("400 GET invalid pageSize: %s", r)
-		err = errors.ErrInvalidParams.WithDetail(
-			fmt.Sprintf("invalid records parameter: %s", err.Error()),
-		)
-		return
-	}
-	if pageSize == 0 {
+	if r != "" {
+		pageSize, err = strconv.ParseInt(r, 10, 32)
+		if err != nil {
+			logger.Errorf("400 GET invalid pageSize: %s", r)
+			err = errors.ErrInvalidParams.WithDetail(
+				fmt.Sprintf("invalid records parameter: %s", err.Error()),
+			)
+			return
+		}
+		if pageSize == 0 {
+			pageSize = notary.DefaultPageSize
+		}
+	} else {
 		pageSize = notary.DefaultPageSize
```

Comment:
@endophage I had the other one drilled into me.  :)  I don't feel particularly strongly.  This is fine too.  We can also do:

```
if r != "" {
  ...
}

if pageSize == 0 {
   pageSize = notary.DefaultPageSize
}
```

(moves the extra check for outside of the `if r != ""` block)